server/lib/availabilitySync.ts

```diff
@@ -0,0 +1,315 @@
+import { getRepository } from 'typeorm';
+import Media from '../entity/Media';
+import { MediaStatus } from '../constants/media';
+import { User } from '../entity/User';
+import type { PlexMetadata } from '../api/plexapi';
+import PlexAPI from '../api/plexapi';
+import logger from '../logger';
+import Season from '../entity/Season';
+import RadarrAPI from '../api/servarr/radarr';
+import type { RadarrSettings, SonarrSettings } from './settings';
+import { getSettings } from './settings';
+import type { SonarrSeason } from '../api/servarr/sonarr';
+import SonarrAPI from '../api/servarr/sonarr';
+
+class AvailabilitySync {
+  public running = false;
+  private plexClient: PlexAPI;
+  private plexSeasonsCache: Record<string, PlexMetadata[]> = {};
+  private sonarrSeasonsCache: Record<string, SonarrSeason[]> = {};
+  private radarrServers: RadarrSettings[];
+  private sonarrServers: SonarrSettings[];
+
+  async run() {
+    const settings = getSettings();
+    this.running = true;
+    this.plexSeasonsCache = {};
+    this.sonarrSeasonsCache = {};
+    this.radarrServers = settings.radarr.filter((server) => server.syncEnabled);
+    this.sonarrServers = settings.sonarr.filter((server) => server.syncEnabled);
+    await this.initPlexClient();
+
+    if (!this.plexClient) {
+      return;
+    }
+
+    logger.debug(`Starting availability sync...`, {
+      label: 'AvailabilitySync',
+    });
+
+    try {
+      const mediaRepository = getRepository(Media);
+      const seasonRepository = getRepository(Season);
+      const availableMedia = await mediaRepository.find({
```

Comment:
Some people have libraries in excess of 30k items. I'm somewhat worried about the performance here. Could we paginate this?